app/serializers/basic_user_serializer.rb

```diff
@@ -1,5 +1,9 @@
 class BasicUserSerializer < ApplicationSerializer
-  attributes :id, :username, :avatar_template
+  attributes :id, :username, :name, :avatar_template
+
+  def name
+    user[:name] rescue nil
```

Comment:
why rescue nil here? this seems not right, also if you look at avatar_template you can see this is a dragon of a nightmare object that is sometimes a Hash and sometimes a User

Reply:
`rescue nil` was not needed per se, I was just trying to be extra cautious here. Thanks for bringing in my notice about the complex case of this object, I amended the code to consider both a Hash and a User object.

Reply:
> rescue nil was not needed per se, I was just trying to be extra cautious here.

Note the `rescue nil` should almost never be used because it rescues any type and all type of errors. I've been bitten multiple times by this where we had typos or the methods that is being called is undefined.